Lint

```
broker.go:137:1: exported function NewAWSAccountBroker should have comment or be unexported
broker.go:137:79: exported func NewAWSAccountBroker returns unexported type main.awsAccountBroker, which can be an
noying to use
```

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -19,6 +19,22 @@ jobs:
           command: dep ensure
 
       - run:
+          name: Run go vet to check for likely mistakes in packages
+          command: go vet
+
+      - run:
+          name: Install goLint
+          command: |
+            go get -u golang.org/x/lint/golint
+            go get -u github.com/GeertJohan/fgt
+
+      # Using fgt to run golint causes the build to fail if any lint is found. 
+      # Otherwise, golint always returns a 0 result.
+      - run:
+          name: Lint checks
+          command: fgt golint
+
+      - run:
           name: Run go test
           command: go test
 

--- a/broker.go
+++ b/broker.go
@@ -132,7 +132,7 @@ func (b awsAccountBroker) LastOperation(ctx context.Context, instanceID, operati
 	return op, err
 }
 
-func NewAWSAccountBroker(baseEmail string, logger lager.Logger, db *gorm.DB) (awsAccountBroker, error) {
+func newAWSAccountBroker(baseEmail string, logger lager.Logger, db *gorm.DB) (awsAccountBroker, error) {
 	mgr, err := newAccountManager()
 	return awsAccountBroker{mgr, baseEmail, logger, db}, err
 }

--- a/main.go
+++ b/main.go
@@ -71,7 +71,7 @@ func main() {
 	}
 	defer db.Close()
 
-	broker, err := NewAWSAccountBroker(baseEmail, logger, db)
+	broker, err := newAWSAccountBroker(baseEmail, logger, db)
 	if err != nil {
 		logger.Fatal("Problem starting broker", err)
 	}

